### ISSUE TITLE
CIP-30: Permission Sets

### ISSUE THREAD

**Issue by Crypto2099** [cardano-foundation/CIPs#557]
As we move as an ecosystem towards a world where things like signing multiple transactions with a single signature are expected behavior for user's ease, I believe we need to introduce the concept of permissions for specific dApps/domains when it comes to CIP-30 wallet connections so that users can granularly control what permissions or access a given dApp has to interact with their wallet.

**Comment by rphair** [comment]
@Crypto2099 good idea to have an issue for this requirement.  I agree it will be important: maybe important enough to create a CPS for it.

@Ryun1 something seems familiar about it... didn't something like this come up when discussing hierarchies of CIP-30 functions?  (Apologies for not being great in this field; just trying to connect with prior discussion in the many PRs we have for CIP-30.)

**Comment by Crypto2099** [comment]
Perhaps but could also be taken a step further and higher to even the "core" CIP-30 functionality such that:
- I am a user who wants to connect my wallet to a site to `signData` with my wallet but DO NOT want to give that website access to create a transaction on my behalf
- I want to give a site the ability to read my wallet state `getUTxO` without the ability to generate and request signature for a transaction on my behalf

etc :)

**Comment by Ryun1** [comment]
So adding optional permissions per endpoints? 

I do quite like this as an idea, but I don't really see a backwards compatible way to integrate this with existing CIP-30 implementations. I think we would be better off pursuing this as an extension, to something like a [CIP-90?](https://github.com/cardano-foundation/CIPs/pull/462/).

The counter argument is that does not sharing public info that useful? because if not then what are they sharing? why connect? - if they are signing (signData or signTx) then they are revealing their public keys where the dApp can index chain for the wallet's public info.

> I am a user who wants to connect my wallet to a site to signData with my wallet but DO NOT want to give that website access to create a transaction on my behalf

The site would be able to identify your wallet from your signature thus allowing them to index chain for the wallet's UTxO set.

**Comment by rphair** [comment]
@Crypto2099 it seems this issue hit a contingency barrier when it looked best incorporated into the next-generation wallet bridge, which never materialised (cc @Ryun1):
* https://github.com/cardano-foundation/CIPs/pull/462

If there is indeed a way of doing it as a truly backward-compatible [`Web-Wallet Bridge`](https://cips.cardano.org/search?q=web-wallet%20bridge) extension, then maybe you will have the bandwidth now to produce a CPS on the subject (as I originally suggested: our default at the time for new problems) or even a CIP if you've had time to think about a solution since.

In either case we could now proceed with review straightaway whenever you might post a CIP or CPS PR to the repo... where I think further discussion would best take place, since that tends to get better quality review than issues (even when not so old as this one).
